src/mul.cpp

```diff
@@ -413,9 +413,13 @@ RCP<const Basic> mul_expand_two(const RCP<const Basic> &a, const RCP<const Basic
                 // The main bottleneck here is the mul(p.first, q.first) command
                 RCP<const Basic> term = mul(p.first, q.first);
                 if (is_a_Number(*term)) {
-                    iaddnum(outArg(coef), rcp_static_cast<const Number>(term));
+                    iaddnum(outArg(coef),
+                        mulnum(mulnum(p.second, q.second), rcp_static_cast<const Number>(term)));
```

Comment:
Here, `p.second * q.second` is ignored. So `(1+2*sqrt(2))*(1+sqrt(2)` is given as `3 + 3*sqrt(2)`